pgrep.c

```diff
@@ -9,24 +9,46 @@ static void deinit_work_threads();
 static int block_all_signals();
 static void handle_signal(int signum);
 static void *run_signal_thread(void *arg);
+static int init_output_fd(void);
+static int destroy_output_fd(void);
+static int install_sigusr2_handler(void);
+static void handle_sigusr2(int signal);
+static int fetch_current_timestamp(char *buf, size_t sz);
+static void *rotate_output_thread_func(void *arg);
 
 static int *avail_pids = NULL;
 static int *attached_pids = NULL;
 static pthread_t *work_threads = NULL;
 static pthread_t signal_thread;
+static pthread_t rotate_output_thread;
 static int avail_pids_count = 0;
 static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
 static pthread_cond_t can_produce = PTHREAD_COND_INITIALIZER;
 static pthread_cond_t can_consume = PTHREAD_COND_INITIALIZER;
+static int should_rotate = 0;
 static int done_pipe[2] = { -1, -1 };
 
+static int output_fd = -1;
```

Comment:
I understand you used a new variable because `context->event_udata->fd` does not have locking around it, right? Why not add the lock around the already existing `fd`?

Reply:
nice catch, done